packages/guides/src/Compiler/NodeTransformers/MenuNodeTransformers/ContentsMenuEntryNodeTransformer.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Compiler\NodeTransformers\MenuNodeTransformers;
+
+use phpDocumentor\Guides\Compiler\CompilerContext;
+use phpDocumentor\Guides\Nodes\Menu\ContentMenuNode;
+use phpDocumentor\Guides\Nodes\Menu\MenuEntryNode;
+use phpDocumentor\Guides\Nodes\Menu\MenuNode;
+use phpDocumentor\Guides\Nodes\Menu\SectionMenuEntryNode;
+use phpDocumentor\Guides\Nodes\Node;
+use Psr\Log\LoggerInterface;
+
+use function assert;
+
+use const PHP_INT_MAX;
+
+final class ContentsMenuEntryNodeTransformer extends AbstractMenuEntryNodeTransformer
+{
+    private const DEFAULT_MAX_LEVELS = PHP_INT_MAX;
+
+    public function __construct(
+        LoggerInterface $logger,
+    ) {
+        parent::__construct($logger);
+    }
+
+    public function supports(Node $node): bool
+    {
+        return $node instanceof MenuNode || $node instanceof SectionMenuEntryNode;
+    }
+
+    /** @return list<MenuEntryNode> */
+    protected function handleMenuEntry(MenuNode $currentMenu, MenuEntryNode $node, CompilerContext $compilerContext): array
+    {
+        if (!$currentMenu instanceof ContentMenuNode) {
+            return [$node];
+        }
+
+        assert($node instanceof SectionMenuEntryNode);
+        $depth = (int) $currentMenu->getOption('depth', self::DEFAULT_MAX_LEVELS - 1) + 1;
+        $documentEntry = $compilerContext->getDocumentNode()->getDocumentEntry();
+        $menuEntry = new SectionMenuEntryNode(
+            $documentEntry->getFile(),
+            $node->getValue() ?? $documentEntry->getTitle(),
+            [],
+            false,
+            1,
+        );
+        $this->addSubSectionsToMenuEntries($documentEntry, $menuEntry, $depth);
+
+        return $menuEntry->getSections();
+    }
+
+    public function getPriority(): int
+    {
+        // After DocumentEntryTransformer
+        return 4500;
+    }
```

Comment:
It would make sense to move this to the parent, as all menu items should be handled at the same prio?

Reply:
No there are different prios

Reply:
Adding the subentries can for example only be done after all globs have been converted to internalmenuentries